robolectric/src/test/java/org/robolectric/fakes/RenderTest.java

```diff
@@ -0,0 +1,305 @@
+      config.orientation = 0;
+    }
+    return config;
+  }
+
+  private static class MyActivity extends Activity {
+    @Override
+    public void onCreate(Bundle params) {
+      super.onCreate(params);
+      LinearLayout layout = new LinearLayout(getApplicationContext());
+      TextView textView = new TextView(getApplicationContext());
+      textView.setId(1);
+      textView.setText("test");
+      layout.addView(textView);
+      setContentView(layout);
+    }
+  }
+
+  @Test
+  public void testSomething() {
+    final String SDK = System.getenv("ANDROID_HOME");
+    File f = new File(SDK + "/platforms/android-21");
+    assertThat(f != null).isEqualTo(true);
+    RenderServiceFactory factory = RenderServiceFactory.create(f);
+    assertThat(factory != null).isEqualTo(true);
+    ResourceRepository projectRes =
+        new ResourceRepository(new FolderWrapper(PROJECT + "/res"), false/*isFramework*/) {
+          @Override
+          protected ResourceItem createResourceItem(String name) {
+            return new ResourceItem(name);
+          }
+        };
+    projectRes.loadResources();
+    // create the rendering config
+    FolderConfiguration config =
+        RenderServiceFactory.createConfig(1280, 800, ScreenSize.XLARGE, ScreenRatio.LONG,
+            ScreenOrientation.PORTRAIT, Density.MEDIUM, TouchScreen.FINGER, KeyboardState.SOFT,
+            Keyboard.QWERTY, NavigationState.EXPOSED, Navigation.NONAV, 21/*api level*/);
+    // create the resource resolver once for the given config.
+    ResourceResolver resources =
+        factory.createResourceResolver(config, projectRes, "Theme", false/*isProjectTheme*/);
+    // create the render service
+    RenderService renderService = factory.createService(resources, config, new ProjectCallback());
+
+
+    try {
+      RenderSession session =
+          renderService.setLog(new StdOutLogger())
+              .setAppInfo("foo", "icon") // optional
+              .createRenderSession("main");
+            // get the status of the render
+      Result result = session.getResult();
+
+      if (result.isSuccess() == false) {
+        // TODO: assert here
+        System.err.println(result.getErrorMessage());
+        System.exit(1);
+      }
+      session.dispose();
+      //Activity activity = buildActivity(MyActivity.class).create().visible().get();
+      //TextView textView = (TextView) activity.findViewById(1);
+      //LinearLayout layout = (LinearLayout)session.getRootViews().get(0).getViewObject();
+      //activity.setContentView(layout);
+      //assertThat(textView).isNotEqualTo(null);
+      //System.out.println("size" + textView.getWidth());
+      // get the image and save it somewhere.
+      BufferedImage image = session.getImage();
+      ImageIO.write(image, "png", new File("./target/maven-archiver/test.png"));
+      // read the views
+      displayViewObjects(session.getRootViews());
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
```

Comment:
Just throws Exception in tests, you want the test to fail if there is an exception, this will pass.